### PR TITLE
Added a --toggle switch to the mark command

### PR DESCRIPTION
**Pull request by Airblader**
Please don't merge yet, I want to add a bunch of testcases for different scenarios. I've tried them manually, but I'd sleep better having tests to rely on.

### CODE REVIEW

**Comment by Airblader** [comment]
I added some test cases. Any interesting corner case I'm missing?

File: src/commands.c [hunk 1]
```diff
@@ -1037,28 +1037,42 @@ void cmd_workspace_name(I3_CMD, char *name) {
...
+            current->con->mark = sstrdup(mark);
+        }
+    }
 
+    DLOG("Clearing all non-matched windows with this mark\n");
     Con *con;
     TAILQ_FOREACH(con, &all_cons, all_cons) {
+        /* Skip matched windows, we took care of them already. */
+        bool matched = false;
+        TAILQ_FOREACH(current, &owindows, owindows)
```
Comment (stapelberg) [inline src/commands.c]: Please add parens to clarify where this loop starts and ends.

**Comment by stapelberg** [comment]
Looks good overall. Can you please address the comment and rebase it onto next?

File: src/commands.c (hunk 1 above)
Comment (Airblader) [inline src/commands.c]: @stapelberg Done and rebased

**Comment by stapelberg** [comment]
Sorry, I just realized I forgot to push my commit :). Can you rebase again please?

**Comment by Airblader** [comment]
@stapelberg Rebased :)
